helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -529,68 +901,88 @@ private void updateCurrentStatesListeners(List<LiveInstance> liveInstances,
 
   private void reset() {
     logger.info("Resetting the routing table.");
-    RoutingTable newRoutingTable = new RoutingTable();
-    _routingTableRef.set(newRoutingTable);
+    for (String key: _routingTableRefMap.keySet()) {
+      PropertyType propertyType = _routingTableRefMap.get(key).get().getPropertyType();
+      if (propertyType == PropertyType.CUSTOMIZEDVIEW) {
+        String stateType = _routingTableRefMap.get(key).get().getStateType();
+        RoutingTable newRoutingTable = new CustomizedViewRoutingTable(propertyType, stateType);
+        _routingTableRefMap.get(key).set(newRoutingTable);
```

Comment:
nit, move this outside the condition block for reducing verbose code.

Reply:
Done.